nbformat/tests/test_sign.py

```diff
@@ -28,6 +30,20 @@ def setUp(self):
     
     def tearDown(self):
         shutil.rmtree(self.data_dir)
+   
+    def test_invalid_db_file(self):
+        invalid_sql_file = os.path.join(self.data_dir, 'invalid_db_file.db')
+        with open(invalid_sql_file, 'w') as tempfile:
+            tempfile.write(u'[invalid data]')
+
+        invalid_notary = sign.NotebookNotary(
+            db_file=invalid_sql_file,
+            secret=b'secret',
+        )
+        invalid_notary.sign(self.nb)
+
+        testpath.assert_isfile(self.data_dir + "/invalid_db_file.db")
+        testpath.assert_isfile(self.data_dir + "/invalid_db_file.db.bak")
```

Comment:
We should always use `os.path.join()` for constructing paths - if `data_dir` ends with `/` (or `\` on Windows), this might go wrong. We already have the first path as `invalid_sql_file`, though, and it's fine to do `+ '.bak'` for the second.